lib/urlwatch/reporters.py

```diff
@@ -735,3 +741,43 @@ def submit(self):
         else:
             logger.debug('Not formatting as Markdown; dependency on markdown2 not met?')
             client_api.send_message(room_id, body_markdown)
+
+
+class XMPPReporter(TextReporter):
+    """Custom XMPP Reporter"""
+    MAX_LENGTH = 262144
+
+    __kind__ = 'xmpp'
+
+    def submit(self):
+        if aioxmpp is None:
+            raise ImportError('Python module "aioxmpp" not installed')
+
+        sender = self.config['sender']
+        password = self.config['password']
+        recipient = self.config['recipient']
+
+        text = '\n'.join(super().submit())
+
+        if not text:
+            logger.debug('Not sending XMPP message (no changes)')
+            return
+
+        for chunk in self.chunkstring(text, self.MAX_LENGTH):
+            asyncio.run(self.submitToXMPP(sender, password, recipient, chunk))
+
+    async def submitToXMPP(self, sender, password, recipient, chunk):
+        jid = aioxmpp.JID.fromstr(sender)
+        client = aioxmpp.PresenceManagedClient(
+            jid, aioxmpp.make_security_layer(password)
+        )
+        recipient_jid = aioxmpp.JID.fromstr(recipient)
+
+        async with client.connected() as stream:
+            msg = aioxmpp.Message(to=recipient_jid, type_=aioxmpp.MessageType.CHAT,)
+            msg.body[None] = chunk
+
+            await stream.send_and_wait_for_sent(msg)
+
+    def chunkstring(self, string, length):
+        return (string[0 + i: length + i] for i in range(0, len(string), length))
```

Comment:
This seems to be the same code as the Telegram reporter, maybe move the `chunkstring` function to the `TextReporter` and use it from both the Telegram reporter and here.

Reply:
..oh and the Slack reporter also has this code duplicated, maybe clean it up there too and move it to `TextReporter`.